### DISCUSSION TITLE
Trouble with json serializing in tauri command

### DISCUSSION THREAD

**Discussion by StepanZagray** [tauri-apps/tauri#11375] [Q&A] [answered]
In rust I have this command: 
```rust
#[tauri::command]
pub fn remove_network(app: AppHandle, network_name: String) -> Result<String, FileError> {
   // Logic
    Ok("Network removed successfully".to_string())
}
```
And on frontend I got this function: 
```javascript
async function remove_network(networkName: string) {
    await invoke('remove_network', { networkName })
}
```

And for some reason I am getting this error on frontend: 
```
Uncaught (in promise) failed serialize json json
```
Which is weird because I have identical invoke for almost identical command that works fine:
```javascript
async function unlink_directory(pathName: string) {
    await invoke('unlink_directory', { pathName })
}
```
```rust
#[tauri::command]
pub fn unlink_directory(app: AppHandle, path_name: String) -> Result<String, FileError> {
    // Logic
    Ok("Directory unlinked successfully".to_string())
}
```

## Update:

I tried a lot of stuff, but still Im getting error specifically on this command
Stuff I tried: 
-renaming argument
-`cargo clean`
-adding other arguments
-removing this command -> compiling -> re-adding command -> compiling

**Comment by FabianLars** [comment, marked as answer]
- Does networkName contain special chars? Or, did you try simple dummy values?
- Does the command actually return the Ok() or does it return an error from whatever `// logic` contains?
- Is that the exact error you're getting? Doesn't sound like something tauri would throw.

**Reply by StepanZagray** [reply]
@tripp (not sure what his nickname on github is) helped me solve this issue on discord. The problem was that I was getting wrong json value because I forgot to change it after I copy paste it from other function. Suprisingly, even though error was on backend I was getting error on frontend, but after some debugging and improved error handling (replace all `?` with `.unwrap()`) I got panic from rust that told me what was wrong
